test/fuzz_api_submissions.py

```diff
@@ -0,0 +1,191 @@
+        boofuzz.s_size(
+            "Body-Content",
+            output_format="ascii",
+            name="Content-Length-Value",
+            fuzzable=False,
+        )
+        boofuzz.s_static("\r\n", "Content-Length-CRLF")
+
+    boofuzz.s_static("\r\n", "Request-CRLF")
+
+    # Add a fuzzable payload
+    with boofuzz.s_block("Body-Content"):
+        boofuzz.s_delim(b"\xD2", name="COSE tag")
+        boofuzz.s_delim(b"\x84", name="CBOR array tag")
+        boofuzz.s_string(
+            "Body content ...", name="Body-Content-Value", max_len=(1 << 20 - 2)
+        )  # 1MB
+
+    test_request = boofuzz.s_get("SubmitAny")
+    session._fuzz_data_logger.log_info(
+        "Number of mutations: {}".format(test_request.num_mutations())
+    )
+    session.connect(test_request)
+    session.fuzz(max_depth=2)
+    session._fuzz_data_logger.log_info(
+        "Number of tests executed: {}".format(session.num_cases_actually_fuzzed)
+    )
+    session._fuzz_data_logger.log_info("Execution speed: {}".format(session.exec_speed))
+
+
+def test_fuzz_api_submissions_cose_payload():
+    """
+    Randomise parts of the cose envelope and do a successfull submission
+    """
+    current_dir = os.path.dirname(__file__)
+
+    session = boofuzz.Session(
+        target=boofuzz.Target(
+            connection=boofuzz.SSLSocketConnection(
+                host="127.0.0.1", port=8000, sslcontext=do_not_verify_tls
+            )
+        ),
+        post_test_case_callbacks=[response_must_be_400()],
+        receive_data_after_each_request=False,
+        check_data_received_each_request=False,
+        receive_data_after_fuzz=False,
+        ignore_connection_issues_when_sending_fuzz_data=False,
+        ignore_connection_ssl_errors=True,
+        reuse_target_connection=False,
+        sleep_time=0.002,
+        web_port=None,
+    )
+
+    # Create a request variable with fuzzable fields
+    boofuzz.s_initialize(name="SubmitCose")
+    with boofuzz.s_block("Request-Line"):
+        boofuzz.s_static("POST /entries HTTP/1.1\r\n")
+        boofuzz.s_static(
+            "User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:65.0) Gecko/20100101 Firefox/65.0\r\n"
+        )
+        boofuzz.s_static(
+            "Accept: text/html,application/json;q=0.9,image/webp,*/*;q=0.8\r\n"
+        )
+        boofuzz.s_static("Accept-Language: en-US,en;q=0.5\r\n")
+        boofuzz.s_static("Accept-Encoding: gzip, deflate\r\n")
+        boofuzz.s_static("Content-Type: application/cose\r\n")
+        # Add claculated content length
+        boofuzz.s_static("Content-Length: ", name="Content-Length-Header")
+        boofuzz.s_size(
+            "Body-Content",
+            output_format="ascii",
+            name="Content-Length-Value",
+            fuzzable=False,
+        )
+        boofuzz.s_static("\r\n", "Content-Length-CRLF")
+    boofuzz.s_static("\r\n", "Request-CRLF")
+
+    filepath = os.path.join(current_dir, "payloads/cts-hashv-cwtclaims-b64url.cose")
```

Comment:
Correction: it's not always the same payload, but we assume that the mutations always cause the registration to fail, probably very early on in the signature verification. It would be more interesting to produce fuzzed payloads that are signed correctly but are expected to break later on, or not expected to break (valid payload with large and weird subjects for example).

Reply:
Yes it is fuzzing a file a bit but that mainly fails due to it being incorrect cbor along with signature verification failure.
I just did not want to invest more time into it because it is not straightforward to do fuzzing and then signing of that fuzzed data. There might be a way to do it with the boofuzz encoder for a given data but the preparation of the payload with boofuzz is fairly complicated given it is cbor we're talking about.
I'll leave it for later, need to get feedback about this implementation from SDL folks anyway.

Reply:
I was looking for ways to produce CBOR with boofuzz, but it doesn't look like it's going to work. The only CBOR fuzzing I can find is https://github.com/fxamacker/cbor-fuzz, which is not directly useable here. I'll try to think of a way to write something that can generate some interesting payloads and integrate with boofuzz, but I agree that this is a good first pass.